HuntHelper/Managers/Hunts/SirenHuntsManager.cs

```diff
@@ -0,0 +1,158 @@
+    {
+        var patchMobs = _sirenHuntsData[patch].MobOrder.Select(patchOrderMobName =>
+            mobs.FirstOrDefault(mob =>
+            {
+                var mobName = mob.Name;
+                if (0 < mob.Instance) mobName += $" {mob.Instance}";
+                return string.Equals(mobName, patchOrderMobName, StringComparison.CurrentCultureIgnoreCase);
+            })
+        ).ToImmutableList();
+
+        if (patchMobs.All(mob => mob == null)) return string.Empty;
+
+        var urlGlyphs = patchMobs.Select(mob =>
+        {
+            if (mob == null)
+                return "-";
+            else
+                return GetNearestSpawnGlyph(patch, mob);
+        });
+
+        return patch + ">" + string.Join("", urlGlyphs);
+    }
+
+    private string GetNearestSpawnGlyph(string patch, HuntTrainMob mob)
+    {
+        var patchMaps = _sirenHuntsData[patch].Maps;
+        if (!patchMaps.ContainsKey(mob.TerritoryID)) return "?";
+
+        return patchMaps[mob.TerritoryID].SpawnPoints
+            .Select(spawnPoint =>
+            {
+                var distSq = (spawnPoint.Item1 - mob.Position).LengthSquared();
+                return (distSq, spawnPoint.Item2);
+            })
+            .MinBy(pair => pair.Item1)
+            .Item2;
+    }
+
+    private void LoadSirenHuntsData()
+    {
+        //directory should exist by default, as that's where spawn and mob data is stored.
+        if (!File.Exists(_sirenHuntsFilePath)) throw new Exception("Siren Hunts file not found ;-;");
+        var deserialized = JsonConvert.DeserializeObject<IDictionary<string, JObject>>(File.ReadAllText(_sirenHuntsFilePath));
+        if (deserialized == null) throw new Exception("Failed to deserialize Siren Hunts data ;-;");
+
+        _sirenHuntsData = deserialized.Select(kv =>
+        {
+            var patchData = kv.Value;
+            var mobOrder = (patchData["MobOrder"] as IList<JToken>)!.Select(token => token.ToString())
+                .ToImmutableList();
+            var maps = (patchData["Maps"] as IDictionary<string, JToken?>)!.Select(kv =>
+            {
+                var map = _mapDataManager.SpawnPointsList
+                    .Find(spawnPoint => string.Equals(spawnPoint.MapName, kv.Key, StringComparison.CurrentCultureIgnoreCase));
+                var spawnPoints = (kv.Value as IDictionary<string, JToken>)!
+                    .Select(kv =>
+                    {
+                        var pos = (kv.Value as JArray)!.Select(v => (float)v).ToList();
+                        return (new Vector2(pos[0], pos[1]), kv.Key);
+                    })
+                    .ToImmutableList();
+
+                return new SirenHuntsMapData(map!.MapID, spawnPoints);
+            }).ToImmutableDictionary(mapData => mapData.MapID);
+
+            return new SirenHuntsPatchData(kv.Key, mobOrder, maps);
+        }).ToImmutableDictionary(patchData => patchData.Name);
+    }
+}
```

Comment:
i.....don't love this. parsing the json this way is hard to read and understand ;-;. but i did this because it allows the json to be very human friendly, which i thought was worth it because updates on patch releases would require touching the json, so it would be nice if it were easier to read.

but i'm very open to alternative suggestions!